wp-remember-the-galleries/wp-remember-the-galleries.php

```diff
@@ -253,19 +253,52 @@ static function enqueue_scripts() {
 	}
 
 	/**
+	 * Send an error message and any additional data
+	 */
+	static function json_error( $message, $data = array() ) {
+		wp_send_json_error( array(
+			'message' => $message,
+			'data' => $data
+		) );
+	}
+
+	/**
 	 * Process AJAX request to save a gallery. Key galleries by title, just
 	 * because it's easier that way and makes it editorially simpler to identify
 	 * different galleries.
 	 */
 	static function save_gallery() {
-		if( !isset( $_POST['images'] ) || !is_array( $_POST['images'] ) ) {
-			wp_send_json_error( 'invalid-input' );
+		$input = filter_input_array( INPUT_POST, array(
+			'images' => array(
+				'filter' => FILTER_VALIDATE_INT,
+				'flags' => FILTER_REQUIRE_ARRAY,
+				'options' => array( 'min_range' => 1 )
+			),
+			'name' => array(
+				'filter' => FILTER_SANITIZE_STRING,
+				'flags' => FILTER_REQUIRE_SCALAR
+			),
+			'term_id' => FILTER_VALIDATE_INT,
+			'settings' => array(
+				'flags' => FILTER_REQUIRE_ARRAY
+			),
+			'yes' => FILTER_VALIDATE_BOOLEAN
+		) );
+
+		extract( $input );
```

Comment:
It's not at all clear what would cause the filter to fail entirely - each individual input value might fail, but there's no description of what causes the "filter to fail", which is mentioned in the docs.